NOTES.md

```diff
@@ -1,3 +1,231 @@
+# Release 2.3.0 (Fantasia)
+
+Released on 2016-04-06
+
+RethinkDB 2.3 introduces a user and permission system, TLS encrypted connections, a
+Windows beta, and various improvements to the ReQL query language.
+
+Read the [blog post][2.3-release] for more details.
+
+[2.3-release]: http://rethinkdb.com/blog/2.3-release/
+
+## Compatibility ##
+
+Data files from RethinkDB version 1.16 onward will be automatically migrated.
+As with any major release, back up your data files before performing the upgrade.
+
+If you're upgrading from RethinkDB 1.15.x or earlier, please read the
+[data migration guide][data-migration-docs] to find out about the required migration steps.
+
+RethinkDB 2.3.0 servers cannot be mixed with servers running RethinkDB 2.2.x or earlier
+in the same cluster.
+
+[data-migration-docs]: http://www.rethinkdb.com/docs/migration/
+
+### API-breaking changes ###
```

Comment:
I should add a bullet point or section for the new `--initial-password` requirement on password-protected clusters... That might otherwise be unexpected for users.